src/impl_pipeline.cpp

```diff
@@ -378,48 +364,75 @@ namespace daxa
         //     }
         // }
 
-        VkShaderModule v_vk_shader_module = {};
-        VkShaderModule p_vk_shader_module = {};
+        u32 neeeded_shader_stages = 0;
```

Comment:
What is this for? You use it to reserve a vector below. Why don't you just append to the vector with the binary WITHIN the first lambda? No reason to pre-reserve here. If it was actually a problem, we could just count up at the beginning, but not separate the work of creating the shader module and putting it into the vector